Sources/CoreBusiness/DataProvider/Server.swift

```diff
@@ -7,29 +7,102 @@
 import Foundation
 
 /// A server environment.
-public enum Server {
+public struct Server {
+#if os(iOS)
+    private static let vector = "appplay"
+#else
+    private static let vector = "tvplay"
+#endif
+
     /// Production.
-    case production
+    public static let production = custom(baseUrl: URL(string: "https://il.srgssr.ch")!)
 
     /// Stage.
-    case stage
+    public static let stage = custom(baseUrl: URL(string: "https://il-stage.srgssr.ch")!)
 
     /// Test.
-    case test
-
-    /// MMF.
-    case mmf
-
-    var url: URL {
-        switch self {
-        case .production:
-            return URL(string: "https://il.srgssr.ch")!
-        case .stage:
-            return URL(string: "https://il-stage.srgssr.ch")!
-        case .test:
-            return URL(string: "https://il-test.srgssr.ch")!
-        case .mmf:
-            return URL(string: "https://play-mmf.herokuapp.com")!
+    public static let test = custom(baseUrl: URL(string: "https://il-test.srgssr.ch")!)
+
+    private let requestBuilder: (String) -> URLRequest
+    private let resizedImageUrlBuilder: (URL, ImageWidth) -> URL
+
+    private init(
+        requestBuilder: @escaping (String) -> URLRequest,
+        resizedImageUrlBuilder: @escaping (URL, ImageWidth) -> URL
+    ) {
+        self.requestBuilder = requestBuilder
+        self.resizedImageUrlBuilder = resizedImageUrlBuilder
+    }
+
+    /// Custom environment.
+    ///
+    /// - Parameters:
+    ///   - requestBuilder: A closure building a playback metadata (media composition) request from a base
+    ///     URL and unique identifier (URN).
+    ///   - resizedImageUrlBuilder: A closure building a resized URL for an input URL and width.
+    ///
+    /// Useful for servers which can pose as SRG SSR servers and deliver the same playback metadata format (and
+    /// image resizing capabilities if possible).
+    public static func custom(
+        requestBuilder: @escaping (String) -> URLRequest,
+        resizedImageUrlBuilder: @escaping (URL, ImageWidth) -> URL
+    ) -> Self {
+        .init(requestBuilder: requestBuilder, resizedImageUrlBuilder: resizedImageUrlBuilder)
+    }
+
+    /// Custom environment.
+    ///
+    /// - Parameters:
+    ///   - baseUrl: The base URL of the server.
+    ///   - queryItems: Additional query items to use.
+    ///
+    /// Useful for servers which can exactly pose as SRG SSR servers and deliver the same playback metadata format and
+    /// image scaling capabilities.
+    public static func custom(baseUrl: URL, queryItems: [URLQueryItem] = []) -> Self {
```

Comment:
What do you think about renaming `custom` by `urn`?

```suggestion
    public static func urn(baseUrl: URL, queryItems: [URLQueryItem] = []) -> Self {
```

Reply:
Would be misleading IMHO. URN is just an identifier for content on a server, not the server itself. Maybe `custom` is not the proper word for it but we need a way to define custom servers which behave like an IntegrationLayer (e.g. MMF or SAM).

Reply:
Should we rename the urn closure parameter by identifier then?

Reply:
I exposed the public initializer instead.